### PR TITLE
Added fleet functionality 

### PR DESCRIPTION
**Pull request by audrastump**
Changes to support setting context for a Fleet resource

### CODE REVIEW

File: src/run.ts [hunk 1]
```diff
@@ -29,6 +29,10 @@ export async function run() {
          required: true
       })
       const clusterName = core.getInput('cluster-name', {required: true})
+      const resourceType = (
```
Comment (davidgamero) [inline src/run.ts:32]: can we add a union type for this value, and some constants to allow comparison?

Comment (davidgamero) [reply src/run.ts:32]:
```typescript
const resourceTypeFleet = "microsoft.containerservice/fleets"
const resourceTypeManagedCluster = "microsoft.containerservice/managedclusters"
type resourceType: resourceTypeFleet | resourceTypeManagedCluster

const resourceInput = core.getInput('resource-type') || 'Microsoft.ContainerService/managedClusters').toLowerCase()
if (
         resourceType !== 'microsoft.containerservice/fleets' &&
         resourceType !== 'microsoft.containerservice/managedclusters'
      ) {
         throw Error(
            'Resource type not recognized, either Microsoft.ContainerService/managedClusters or Microsoft.ContainerService/fleets is valid'
         )
}
const resourceType: resourceType = resourceInput // now we can use the consts for comparison

**Comment by bfoley13** [comment]
Add to the description of this PR as well

File: src/run.ts [hunk 2]
```diff
@@ -37,6 +41,28 @@ export async function run() {
       const publicFqdnInput = core.getInput('public-fqdn') || ''
       const publicFqdn = publicFqdnInput.toLowerCase() === 'true'
 
+      // check resource type is recognized
+      if (
+         resourceType !== 'microsoft.containerservice/fleets' &&
+         resourceType !== 'microsoft.containerservice/managedclusters'
```
[the remark this answers is not in the document]
Comment (bfoley13) [reply src/run.ts]: I totally missed the `.toLowerCase` on the input, looks good

**Review by bfoley13 — APPROVED** [review]

File: src/run.ts [hunk 3]
```diff
@@ -29,6 +29,17 @@ export async function run() {
...
+      const resourceTypeFleet = 'microsoft.containerservice/fleets'
+      const resourceTypeManagedCluster =
+         'microsoft.containerservice/managedclusters'
+      type ResourceType =
+         | typeof resourceTypeFleet
+         | typeof resourceTypeManagedCluster
+      const resourceInput = (
+         core.getInput('resource-type') ||
+         'Microsoft.ContainerService/managedClusters'
+      ).toLowerCase() as ResourceType
```
Comment (davidgamero) [inline src/run.ts:41]: we shouldn't need this type assertion if the narrowing happens before we assign to the ResourceType:

1. ResourceInput is read as a string
2.  We throw if it has values that are not either `typeof resourceTypeFleet` or `typeof resourceTypeCluster`
3.  Now the type is narrowed to fit the `type ResourceType` so we can directly assign without `as`

**Review by davidgamero — APPROVED** [review]
lgtm

**Review by bfoley13 — APPROVED** [review]

**Pull request merged**